src/server.php

```diff
@@ -501,9 +503,15 @@ public function search($param) {
           $work_ids[] = [$uid];
         }
       } else {
-        $this->cache = new cache($this->config->get_value('cache_host', 'setup'),
-          $this->config->get_value('cache_port', 'setup'),
-          $this->config->get_value('cache_expire', 'setup'));
+        if ($this->config->get_value('cache_type', 'setup') == "redis") {
+          $this->cache = new rediscache($this->config->get_value('cache_host', 'setup'),
+            $this->config->get_value('cache_port', 'setup'),
+            $this->config->get_value('cache_expire', 'setup'));
+        } else {
+          $this->cache = new cache($this->config->get_value('cache_host', 'setup'),
+            $this->config->get_value('cache_port', 'setup'),
+            $this->config->get_value('cache_expire', 'setup'));
+        }
```

Comment:
Look like this code block is repeated below?

Reply:
Yup. It was like that originally. Not going to mess with that. :-(